NYPL-Simplified/circulation
# Fix RBDigital catalog delta sync

## JIRA Ticket

https://jira.nypl.org/browse/SIMPLY-2747

## What does this PR do?

This PR addresses an issue caused by changes to the RBDigital API.

Specifically, the `/v1/libraries/:libraryId/media/delta` endpoint, has been deprecated. It previously worked by accepting `begin` and `end` date query parameters, from which it would calculate and return the delta between those dates and return cataloging metadata for the items involved.

In the new version,
- the `begin` and `end` dates are selected from a list of snapshots from the `/v1/libraries/:libraryId/book-holdings/delta/available-dates` endpoint,
- those dates are then given as query parameters to the `/v1/libraries/:libraryId/book-holdings/delta` endpoint (the format of this response has changed, as well), and finally
- the ISBN for each added item is used as the `:token` for the `/v1/libraries/:libraryId/media/:token` endpoint to retrieve the catalog metadata in the correct format.

The PR also fixes a couple of bugs on apparently seldomly traversed paths.

## How should this be tested?

Run the `bin/rbdigital_collection_delta` script in QA, test, and/or dev environment and ensure that it completes successfully.

Run test suite, which has been updated with new tests, assertions, and content that reflect added and changed methods and new API endpoint responses.

## Who has tested?
@tdilauro - In two dev environments from dumps of different Circulation Manager instances.
